Yank/multistate/multistateanalyzer.py

```diff
@@ -871,48 +919,43 @@ def show_mixing_statistics(self, cutoff=0.05, number_equilibrated=None):
         logger.info('Replica state index statistical inefficiency is '
                     '{:.3f}'.format(mixing_statistics.statistical_inefficiency))
 
-    def get_timeseries_weights(self, sampled_states_at_iteration, iteration):
+    def get_effective_energy_timeseries(self):
         """
-        Return the weights of the timeseries from disk
+        Generate the effective energy (negative log deviance) timeseries that is generated for this phase
 
-        For now this is just a placeholder, but may be more complex in the future
-        """
-        return np.zeros(sampled_states_at_iteration.shape)
+        The effective energy for a series of samples x_n, n = 1..N, is defined as
 
-    def get_timeseries(self, passed_timeseries, sampled_states):
-        """
-        Compute the timeseries of a simulation from the Replica Exchange simulation. This is the sum of energies
-        for each sample from the state it was drawn from.
+        u_n = - \ln \pi(x_n) + c
 
-        Parameters
-        ----------
-        passed_timeseries : np.ndarray of shape (K,L,N), indexed by k,l,n
-            K is the total number of sampled states
+        where \pi(x) is the probability density being sampled, and c is an arbitrary constant.
 
-            L is the total states we want MBAR to analyze
+        Returns
+        -------
+        u_n : ndarray of shape (N,)
+            u_n[n] is the negative log deviance of the same from iteration ``n``
+            Timeseries used to determine equilibration time and statistical inefficiency.
 
-            N is the total number of samples
+        """
+        energies, _, _, states = self.read_energies()
+        n_replicas, n_states, n_iterations = energies.shape
 
-            The kth sample was drawn from state k at iteration n, the nth configuration of kth state is evaluated in
-            thermodynamic state l
+        # Check for log weights
+        has_log_weights = False
+        if self.has_log_weights:
+            has_log_weights = True
+            log_weights = self.read_log_weights()
+            f_l = - self.read_logZ(iteration=-1) # use last (best) estimate of free energies
```

Comment:
Note to fix later: the logic test here and then again later seems redundant, we can find a way to clean it up later